packages/ihe-gateway-sdk/src/models/document-query.ts

```diff
@@ -1,34 +1,68 @@
-import { BaseRequest, baseResponseSchema, documentReference } from "./shared";
+import {
+  BaseRequest,
+  baseRequestSchema,
+  baseResponseSchema,
+  BaseResponse,
+  documentReferenceSchema,
+  DocumentReference,
+  XCAGateway,
+} from "./shared";
 import { z } from "zod";
 
-export type Code = {
-  system: string;
-  code: string;
-};
+export const codeSchema = z.object({
+  system: z.string(),
+  code: z.string(),
+});
 
-export type DateRange = {
-  dateFrom: string;
-  dateTo: string;
-};
+export const dateRangeSchema = z.object({
+  dateFrom: z.string(),
+  dateTo: z.string(),
+});
+
+export const xcpdPatientIdSchema = z.object({
+  id: z.string(),
+  system: z.string(),
+});
+
+export type XCPDPatientId = z.infer<typeof xcpdPatientIdSchema>;
+export type Code = z.infer<typeof codeSchema>;
+export type DateRange = z.infer<typeof dateRangeSchema>;
 
-export type DocumentQueryRequest = BaseRequest & {
-  xcaHomeCommunityId: string;
-  xcpdPatientId: {
-    id: string;
-    system: string;
-  };
+// The following are for us creating a document query request
+
+export type DocumentQueryRequestOutgoing = BaseRequest & {
+  cxId: string;
+  xcpdPatientId: XCPDPatientId;
   patientId?: string;
-  xcaGateway: string;
```

Comment:
updated in line with current defintiion:

"gateway": {
     "homeCommunityId": "1.2.840.114350.1.13.11511.3.7.3.688884.100.1000",
     "url": "https://sfd-np.et0121.epichosted.com:14430/interconnect-ce-env1/wcf/epic.community.hie/respondinggateway.svc/ceq"
  },